### PR TITLE
Automatic loading of default configuration in `TelestionVerticle`

### PR DESCRIPTION
**Pull request by fussel178**
### Summary 

Adds functionaltiy to `TelestionVerticle` that tries to load the default configuration from the configuration class.

### Details 

- add two constructors to `TelestionVerticle` which try load the default configuration
- add an example which shows the usage of the "autoload" feature of the `TelestionVerticle`

### Additional information 

During the migration in the https://github.com/wuespace/telestion-project-daedalus2, I always load the default configuration through the default constructor of the configuration class, like so:

```java
public class SomeVerticle extends TelestionVerticle<SomeVerticle.Configuration> {

	public record Configuration(
		@JsonProperty String someProperty
	) implements TelestionConfiguration {
		public Configuration() {
			this("Some default configuration. Feel free to overwrite me!");
		}
	}

	@Overwrite
	public void onStart() {
		setDefaultConfiguration(new Configuration());
		// use normal configuration, do something else, etc.
	}
}
```

Well,  that's repetitive. Let the base class do that, I think. :upside_down_face: 

And here is the new example:
https://github.com/wuespace/telestion-core/blob/1061e43228f2d857e1867c45e74cad119e3fa916/modules/telestion-examples/src/main/java/de/wuespace/telestion/examples/DefaultConfigVerticle.java

If you don't like this feature, you can turn it off in the constructor.

As always, I'm open for suggestions. :wink: 

### CLA

- [x] I have signed the individual contributor's license agreement **and sent** it to the board of the WüSpace e. V. organization.

### CODE REVIEW

File: modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/TelestionVerticle.java [hunk 1]
```diff
@@ -60,6 +62,32 @@
...
+	 */
+	public TelestionVerticle(boolean skipDefaultConfigLoading) {
+		if (skipDefaultConfigLoading) return;
+		try {
+			var defaultConfig = getConfigType().getConstructor().newInstance();
+			this.defaultConfig = defaultConfig;
+			this.defaultGenericConfig = defaultConfig.json();
+		} catch (NoSuchMethodException | InvocationTargetException | InstantiationException | IllegalAccessException e) {
+			// no default configuration on configuration class found, ignoring
+			logger.debug("No default configuration found. Skip loading");
```
Comment (pklaschka) [inline modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/TelestionVerticle.java:79]: Let's make this message a little more expressive, something along the lines of 

> No default configuration found in configuration class `...`. Expected default constructor with no arguments to exist on `ConfigurationRecordClass`. Continuing without default configuration.

**Review by pklaschka — CHANGES_REQUESTED** [review]
Overall, I like it.

The only thing (I left an inline comment) would be the log message when no default configuration exists. I would suggest using [an AWESA message](https://wuespace.github.io/telestion-docs/internal/references/actionable-error-message-structure/) here, given that we do expect this to happen and can therefore provide actionable messages (which can be interpreted as "error messages" if the developer's intention had a default config).

File: modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/TelestionVerticle.java (hunk 1 above)
Comment (fussel178) [reply modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/TelestionVerticle.java]: Good point!
Done in [`0faeb44` (#466)](https://github.com/wuespace/telestion-core/pull/466/commits/0faeb448b63515961a118203ef2bf17870c204c2).

I also changed the log type to `info`. I think that makes more sense.

**Review by pklaschka — DISMISSED** [review]
LGTM. Thank you for your contribution 🎉 🙂

**Review by cb0s — DISMISSED** [review]
✔️

File: modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/TelestionVerticle.java [hunk 2]
```diff
@@ -60,6 +63,38 @@
...
 	}
 
+	/**
+	 * Creates a new Telestion verticle and tries to load the default configuration
+	 * from the specified configuration class.
+	 *
+	 * @param skipDefaultConfigLoading when {@code true} the loading of the default configuration is skipped
+	 */
+	public TelestionVerticle(boolean skipDefaultConfigLoading) {
+		if (skipDefaultConfigLoading) return;
```
Comment (cb0s) [inline modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/TelestionVerticle.java:73]:
```suggestion
		if (skipDefaultConfigLoading) {
			return;
		}
```

File: modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/TelestionVerticle.java [hunk 3]
```diff
@@ -60,6 +63,38 @@
...
+		var configType = getConfigType();
+		if (Objects.isNull(configType)) return;
```
Comment (cb0s) [inline modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/TelestionVerticle.java:75]:
```suggestion
		if (Objects.isNull(configType)) {
			return;
		}
```

**Review by cb0s — CHANGES_REQUESTED** [review]
We usually don't use the one liner ifs as they can be harder to read in bigger chunks of code. It is just a small improvement in my opinion so if you do not agree with me in this point, you can merge it without accepting my proposals. :)

**Pull request merged**